packages/embedded-wallet/src/ui/slider/Slider.test.tsx

```diff
@@ -0,0 +1,129 @@
+  };
+};
+
+const triggerKeyOn = (component: MockSlider, element: keyof MockSlider, key: string) => {
+  component[element].simulate('keydown', {key});
+  return refreshSliderFrom(component);
+};
+
+describe('UI - Slider', () => {
+  let component: MockSlider;
+
+  beforeEach(() => {
+    component = mockSlider();
+  });
+
+  it('can be instantiated', () => {
+    expect(component.minLabelElement.exists()).toEqual(true);
+    expect(component.minLabelElement.text()).toEqual('0');
+    expect(component.minLabelElement.prop('aria-hidden')).toEqual('true');
+    expect(component.valueLabelElement.exists()).toEqual(true);
+    expect(component.valueLabelElement.text()).toEqual('0');
+    expect(component.valueLabelElement.prop('aria-hidden')).toEqual('true');
+    expect(component.maxLabelElement.exists()).toEqual(true);
+    expect(component.maxLabelElement.text()).toEqual('100ETH');
+    expect(component.maxLabelElement.prop('aria-hidden')).toEqual('true');
+    expect(component.sliderElement.exists()).toEqual(true);
+    expect(component.sliderElement.hasClass(css.slider)).toEqual(true);
+    expect(component.sliderElement.prop('min')).toEqual(0);
+    expect(component.sliderElement.prop('max')).toEqual(100);
+    expect(component.sliderElement.prop('value')).toEqual(0);
+    expect(component.sliderElement.prop('step')).toEqual(1);
+    expect(component.sliderElement.prop('aria-atomic')).toEqual(true);
+    expect(component.sliderElement.prop('aria-valuemin')).toEqual(0);
+    expect(component.sliderElement.prop('aria-valuemax')).toEqual(100);
+    expect(component.sliderElement.prop('aria-valuenow')).toEqual(0);
+    expect(component.sliderElement.prop('aria-label')).toEqual('Current allocation: 0 ETH');
+    expect(component.sliderElement.prop('type')).toEqual('range');
+    expect(component.sliderElement.prop('onChange')).toBeInstanceOf(Function);
+    expect(component.sliderElement.prop('onKeyDown')).toBeInstanceOf(Function);
+  });
+
+  describe('can change its value with the keyboard', () => {
+    const keyboardCases = [
+      ['increasing', 'ArrowUp', 1],
+      ['increasing', 'ArrowRight', 1],
+      ['increasing', 'PageUp', 10],
+      ['decreasing', 'ArrowDown', -1],
+      ['decreasing', 'ArrowLeft', -1],
+      ['decreasing', 'PageDown', -10]
+    ];
+
+    beforeEach(() => {
+      component = mockSlider({initialValue: 50});
+    });
+
+    it.each(keyboardCases)('%s it by %i when pressing %s', (_, key, numberChange) => {
+      component = triggerKeyOn(component, 'sliderElement', key as string);
+      const value = component.sliderElement.prop('value');
+
+      expect(value).toEqual(50 + (numberChange as number));
+      expect(component.valueLabelElement.text()).toEqual(`${value}`);
+      expect(component.sliderElement.prop('aria-label')).toEqual(
+        `Current allocation: ${value} ETH`
+      );
+    });
+  });
+
+  it("can't surpass its maximum value", () => {
+    const changed = jest.fn();
```

Comment:
I... don't understand what this does, 'cause according to [this](https://jestjs.io/docs/en/mock-functions.html#using-a-mock-function), you need to pass an argument to this function so that jest mocks it, so.... I don't get it

Reply:
The purpose of this test is to make sure that an event _is not_ triggered. Since I don't care what the callback does and I just want to know if it was called, I can just spawn an empty mock function and leverage [the `.mock` property](https://jestjs.io/docs/en/mock-functions.html#mock-property) available to all mock functions, and then use matchers such as `toHaveBeenCalled`.